src/main/java/org/jenkins/tools/test/hook/VersionValidationHook.java

```diff
@@ -0,0 +1,21 @@
+package org.jenkins.tools.test.hook;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.apache.maven.model.Model;
+import org.jenkins.tools.test.exception.PluginSourcesUnavailableException;
+import org.jenkins.tools.test.model.hook.BeforeCheckoutContext;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeCheckout;
+import org.kohsuke.MetaInfServices;
+
+@MetaInfServices(PluginCompatTesterHookBeforeCheckout.class)
+public class VersionValidationHook extends PluginCompatTesterHookBeforeCheckout {
+    @Override
+    public void action(@NonNull BeforeCheckoutContext context)
+            throws PluginSourcesUnavailableException {
+        Model model = context.getModel();
+        if (model.getScm().getTag() == null || model.getScm().getTag().equals("HEAD")) {
```

Comment:
I have a feeling this breaks `--local-checkout-dir` for regular (non multi-module) plugins which is a used developer flow to check compatability without having to constantly rebuild a war?

Reply:
Not sure. `--local-checkout-dir` doesn't really work for the local development use case anyway since PCT deletes the whole Git repository before cloning: I'd be too afraid to point it to my local directory out of fear of losing all my local branches. And in automation it's not used at all in `jenkinsci/bom` and used more or less in ignorance of the correct approach in proprietary consumers.

I'm inclined to let it break for now. Once the dust has settled with multimodule support, which I think should include ripping `--local-checkout-dir` from all automation, I'll look into getting it to work again in an intuitive fashion for the local development use case.